src/functions-and-arrays.js

```diff
@@ -29,10 +181,54 @@ const wordsUnique = [
   'bring'
 ];
 
+function uniquifyArray(arrayWords){
+
+  if(arrayWords.length === 0){
+    return null;
+  }
+
+  // declaramos variables de control
+  let newArray = [];
+
+  // loop para checkear items duplicados y dejar sólo uno en el newArray
+  for(let i = 0; i < arrayWords.length; i++){
+    if(newArray.indexOf(arrayWords[i]) === -1){
+      newArray.push(arrayWords[i]);
+    }
+  }
+
+  return newArray;
+
+};
+
 // Iteration #6: Find elements
+// -- Let's create a simple array search.
+// -- Declare a function named doesWordExist that will take in an array of words as one argument, and a word to search for as the other. 
+// -- Return true if it exists, otherwise, return false. Don't use indexOf for this one.
+
 const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
 
+function doesWordExist(arrayWords, wordSearch){
+
+  if(arrayWords.length === 0){
+    return null;
+  }
+
+  // loop para chequear si el item existe en el array usando method arr.includes()
+
```

Comment:
Estas seguro de que esta funcion paso todos sus tests?

Creo que el return false lo queres poner afuera del loop

Reply:
Sí, seguro. Jasmine me indica 4 failures (3 del Bonus Quest y 1 del throw error en el Bonus: Calculate sum).

Porqué lo pondrías afuera del loop?

Reply:
Vamos a seguir el recorrido de la funcion.
Entra al loop, agarra la primera palabra del array, si el if se cumple, entra y la funcion devuelve true! lo cual todo correcto. Ahora si no se cumple entra en el else y devuelve false. Pero devolvio false y solo chequeo la primer palabra, todo el resto del array no lo reviso, es eso lo que querias?

Reply:
Claro, debería recorrer todo el array.. ahora lo entiendo.
Lo que no veo es porqué si el return false está dentro del loop, me sigue dando true la función si busco un valor eg. eating, que no es el primer item del array...

Reply:
Perdon, no preste atencion al metodo que estabas usando para comparar.
Lo que esta de mas es el for loop.
Esto se resuelve asi:

`
if(arrayWords.includes(wordSearch)){
      return true;
    } 
      return false;
   
`
Listo